File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/StepBase.cpp
// StepBase.cpp 
#include "StepBase.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

StepBase::StepBase(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //stepDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* stepDateNode = xmlNode->FirstChildElement("stepDate");

   if(stepDateNode){stepDateIsNull_ = false;}
   else{stepDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- stepDateNode , address : " << stepDateNode << std::endl;
   #endif
   if(stepDateNode)
   {
      if(stepDateNode->Attribute("href") || stepDateNode->Attribute("id"))
      {
          if(stepDateNode->Attribute("id"))
          {
             stepDateIDRef_ = stepDateNode->Attribute("id");
             stepDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(stepDateNode));
             stepDate_->setID(stepDateIDRef_);
             IDManager::instance().setID(stepDateIDRef_,stepDate_);
          }
          else if(stepDateNode->Attribute("href")) { stepDateIDRef_ = stepDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { stepDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(stepDateNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeDate> StepBase::getStepDate()
{
   if(!this->stepDateIsNull_){
        if(stepDateIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDate>(IDManager::instance().getID(stepDateIDRef_));
        }else{
             return this->stepDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDate>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/StepBase.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class StepBase : ISerialized
    {
        public StepBase(XmlNode xmlNode)
        {
            XmlNode stepDateNode = xmlNode.SelectSingleNode("stepDate");
            
            if (stepDateNode != null)
            {
                if (stepDateNode.Attributes["href"] != null || stepDateNode.Attributes["id"] != null) 
                {
                    if (stepDateNode.Attributes["id"] != null) 
                    {
                        stepDateIDRef_ = stepDateNode.Attributes["id"].Value;
                        XsdTypeDate ob = new XsdTypeDate(stepDateNode);
                        IDManager.SetID(stepDateIDRef_, ob);
                    }
                    else if (stepDateNode.Attributes["href"] != null)
                    {
                        stepDateIDRef_ = stepDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        stepDate_ = new XsdTypeDate(stepDateNode);
                    }
                }
                else
                {
                    stepDate_ = new XsdTypeDate(stepDateNode);
                }
            }
            
        
        }
        
    
        #region StepDate_
        private XsdTypeDate stepDate_;
        public XsdTypeDate StepDate_
        {
            get
            {
                if (this.stepDate_ != null)
                {
                    return this.stepDate_; 
                }
                else if (this.stepDateIDRef_ != null)
                {
                    stepDate_ = IDManager.getID(stepDateIDRef_) as XsdTypeDate;
                    return this.stepDate_; 
                }
                else
                {
                      return this.stepDate_; 
                }
            }
            set
            {
                if (this.stepDate_ != value)
                {
                    this.stepDate_ = value;
                }
            }
        }
        #endregion
        
        public string stepDateIDRef_ { get; set; }
        
    
        
    
    }
    
}

